packages/react-reconciler/src/ReactFiberListenerEvents.js

```diff
@@ -0,0 +1,244 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import type {Fiber, Interactions} from './ReactFiber';
+import type {Container, ReactListenerType} from './ReactFiberHostConfig';
+import type {ReactListenerInstance} from 'shared/ReactTypes';
+
+import {NoWork} from './ReactFiberExpirationTime';
+import {
+  mountListenerInstance,
+  unmountListenerInstance,
+} from './ReactFiberHostConfig';
+
+import {HostComponent, HostRoot} from 'shared/ReactWorkTags';
+
+const isArray = Array.isArray;
```

Comment:
This doesn't seem necessary given it's used once in the file

Reply:
It's to improve perf on having to get Array.isArray vs caching it for a hot function below.

Reply:
Do JS VMs really not have optimisations for that? I find it hard to believe there will be any different

Reply:
We do it elsewhere in the React codebase for that reason too. The other reason is so we can cache Array.isArray before a polyfill might get added, which can lead to de-opts, but that's a separate topic really.